src/v/cluster/partition_balancer_planner.cc

```diff
@@ -162,37 +162,42 @@ void partition_balancer_planner::init_per_node_state(
             vlog(clusterlog.debug, "node {}: decommissioning", id);
             ctx.decommissioning_nodes.insert(id);
         }
-    }
-
-    const auto now = raft::clock_type::now();
-    for (const auto& follower : follower_metrics) {
-        auto unavailable_dur = now - follower.last_heartbeat;
+        auto node_status = _state.node_status().get_node_status(id);
+        // node status is not yet available, wait for it to be updated
```

Comment:
As discussed on slack, we should cover the case when a remote node didn't respond to pings at all since the current node startup. I guess we should use time when we discovered the node through the members table as last_seen in this case.

Reply:
i will open a follow up pr with the changes in `node_status` subsystem it seems that we also do not clean the status table when node is removed from the cluster